user_guide.adoc

```diff
@@ -3878,19 +3880,19 @@ specify that, but instead one can use:
 * `bool` to specify booleans
 * `struct` to allow arbitrary KaitaiStruct-compatible user types
 * `io` to specify KaitaiStream-compatible IO streams
-* `any` to allow any type (if target language supports that)
+* `any` to allow any type (if the target language supports that)
 
 NOTE: Parametric types with mandatory parameters can't be read from
-file directly (as they obviously need parameter values supplied from
-somewhere), and thus typical `fromFile(...)` helper method is not
+files directly (as they obviously need parameter values supplied from
```

Comment:
I would actually prefer the singular form, i.e. _from **a** file_ or _from **the** file_. "Reading from files" might sound like we want to read a single type from several files at once, which is not the case.

Reply:
Yep, agreed. Done.